cloud/ec2.go

```diff
@@ -428,16 +428,9 @@ func (m *ec2Manager) SpawnHost(ctx context.Context, h *host.Host) (*host.Host, e
 	if err = ec2Settings.Validate(); err != nil {
 		return nil, errors.Wrapf(err, "invalid EC2 settings in distro %s: %+v", h.Distro.Id, ec2Settings)
 	}
-	if ec2Settings.KeyName == "" && !h.UserHost {
-		if !h.SpawnOptions.SpawnedByTask {
-			return nil, errors.New("key name must not be empty")
-		}
-		var k string
-		k, err = m.client.GetKey(ctx, h)
-		if err != nil {
-			return nil, errors.Wrap(err, "getting key name")
-		}
-		ec2Settings.KeyName = k
+	ec2Settings.KeyName, err = getKeyName(ctx, h, m.settings, m.client)
```

Comment:
I know this isn't new, but would it make sense to add a comment along the lines of 
// the KeyName will be used by AWS to determine which public key to put on the host

Reply:
done